src/clients/uploads-api-client.ts

```diff
@@ -0,0 +1,36 @@
+import { amazonMarketplaces } from '@scaleleap/amazon-marketplaces'
+import globalAxios, { AxiosError, AxiosInstance, AxiosResponse } from 'axios'
+
+import { UploadsApi } from '../api-models/uploads-api-model'
+import { apiErrorFactory } from '../helpers'
+import { isJsonMime } from '../helpers/is-json-mime'
+import { APIConfigurationParameters } from '../types/api-configuration-parameters'
+
+export class UploadsApiClient extends UploadsApi {
+  constructor(parameters?: APIConfigurationParameters, axios?: AxiosInstance) {
```

Comment:
We don't need to allow end consumers to swap out `axios`. `axios` is an implementation detail.

Reply:
It might be necessary for testing? We can mock an `axios` instance?

Reply:
Ok, sure. Then why not make it an optional property on `APIConfigurationParameters`? Why does it need to be a separate parameter?